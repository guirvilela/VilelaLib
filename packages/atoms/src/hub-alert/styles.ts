import styled from 'styled-components';

import { HubAlertProps } from './types';

export const AlertInnerStyle = styled.div`
  display: flex;
  gap: ${({ theme }) => theme.spaces[2]};
  margin-top: -2px;
  width: 100%;
`;

export const AlertTitleStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutral[800]};
`;

export const AlertContentStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutralGrey[700]};
  font-weight: 400;
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
`;

export const AlertCloseStyle = styled.button`
  background-color: transparent;
  width: ${({ theme }) => theme.spaces[3]};
  height: ${({ theme }) => theme.spaces[3]};
  border: none;
  color: inherit;
  cursor: pointer;
`;

export const AlertStyle = styled.div<Omit<HubAlertProps, 'title'>>`
  display: flex;
  width: 100%;
  background-color: ${({ theme, color }) =>
    color ? theme.colors[color][100] : theme.colors.neutral[100]};
  color: ${({ theme, color }) =>
    color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border: 1px solid
    ${({ theme, color }) =>
      color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border-radius: ${({ theme, radius }) =>
    radius ? theme.spaces[radius] : '4px'};
  align-items: ${({ inline }) => (inline ? 'center' : 'start')};

  ${({ theme, pad }) => {
    if (typeof pad === 'string') {
      return `padding: ${theme.spaces[pad]};`;
    }
    if (Array.isArray(pad)) {
      return `padding: ${pad.reduce(
        (acc, x) => acc + `${theme.spaces[x]} `,
        ''
      )};`;
    }
    return `padding: ${theme.spaces[2]};`;
  }}

  gap: ${({ theme, gap }) => (gap ? theme.spaces[gap] : theme.spaces[2])};

  ${AlertInnerStyle} {
    flex-direction: ${({ inline }) => (inline ? 'row' : 'column')};
    align-items: ${({ inline }) => (inline ? 'center' : 'start')};
  }

  ${AlertTitleStyle} {
    font-weight: ${({ inline }) => (inline ? 400 : 600)};
    white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : undefined)};
  }
`;
