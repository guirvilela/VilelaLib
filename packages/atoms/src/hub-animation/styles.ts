import styled from 'styled-components';

import { HubAnimationProps } from './types';

export const AnimationStyle = styled.div<HubAnimationProps & { ref }>`
  ${({ theme, size, width, height }) => {
    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
          min-width: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[6]};
          height: ${theme.spaces[6]};
          min-width: ${theme.spaces[6]};
        `;
      default:
        return `
          width: ${width || theme.spaces[5]};
          height: ${height || theme.spaces[5]};
          min-width: ${width || theme.spaces[5]};
        `;
    }
  }}
`;
