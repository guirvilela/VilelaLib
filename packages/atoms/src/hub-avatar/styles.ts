import styled from 'styled-components';

import { HubAvatarProps } from './types';

export const AvatarImgStyle = styled.img`
  width: 100%;
  height: 100%;
  background-size: cover;
  border-radius: 50%;
`;

export const AvatarInitialsStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

export const AvatarStyle = styled.div<HubAvatarProps>`
  ${({ theme, size }) => {
    if (!size) {
      return `
        width: ${theme.spaces[7]};
        height: ${theme.spaces[7]};
        min-width: ${theme.spaces[7]};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[7]};
          height: ${theme.spaces[7]};
          min-width: ${theme.spaces[7]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[10]};
          height: ${theme.spaces[10]};
          min-width: ${theme.spaces[10]};
        `;
      default:
        return `
          width: ${size};
          height: ${size};
          min-width: ${size};
        `;
    }
  }}
  ${({ borderless, theme }) =>
    borderless
      ? ''
      : `
    padding: ${theme.spaces[1]};
    border: 1px solid ${theme.colors.neutral[400]};
  `};

  border-radius: 50%;

  ${AvatarInitialsStyle} {
    ${({ theme, typo }) => `
      font-family: ${theme.typography[typo || 'h4'].family};
      font-size: ${theme.typography[typo || 'h4'].size};
      line-height: ${theme.typography[typo || 'h4'].lineHeight};
      letter-spacing: ${theme.typography[typo || 'h4'].letterSpacing};
    `}

    background-color: ${({ theme, color, colorLevel }) =>
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : undefined};

    color: ${({ theme, colorText, colorTextLevel }) =>
      colorText && colorTextLevel
        ? theme.colors[colorText][colorTextLevel]
        : colorText && theme.colors[colorText]
        ? theme.colors[colorText][500]
        : colorText
        ? colorText
        : theme.colors.neutral[800]};
  }
`;
