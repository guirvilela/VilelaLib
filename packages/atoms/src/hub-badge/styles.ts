import styled, { css } from 'styled-components';

import { HubBadgeProps } from './types';

const allStyle = css<HubBadgeProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: ${({ theme }) => theme.spaces[2]};

  ${({ theme, size, width, height }) => {
    if (width || height) {
      return `
        width: ${width};
        height: ${height};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[4]};
          height: ${theme.spaces[4]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
        `;
      default:
        return '';
    }
  }}

  background-color: ${({ theme, color, colorLevel }) =>
    color && colorLevel
      ? theme.colors[color][colorLevel]
      : color
      ? theme.colors[color][100]
      : theme.colors.neutral[200]};

  color: ${({ theme, color, textColor, textColorLevel }) =>
    textColor && textColorLevel
      ? theme.colors[textColor][textColorLevel]
      : textColor
      ? theme.colors[textColor][500]
      : color
      ? theme.colors[color][500]
      : theme.colors.neutral[500]};

  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || '1']};

  ${({ theme, pad }) => {
    if (!pad) return `padding: 2px 6px`;
    if (typeof pad === 'string') return `padding: ${theme.spaces[pad]}`;
    if (pad.length === 4) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      } ${theme.spaces[pad[3]]}`;
    }
    if (pad.length === 3) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      }`;
    }
    if (pad.length === 2) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]}`;
    }
    return `padding: ${theme.spaces[pad[0]]}`;
  }};

  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  font-weight: 600;
`;

const floatStyle = css<HubBadgeProps>`
  position: relative;

  &::after {
    ${allStyle}

    position: absolute;
    content: '${({ content }) => content}';
    border: 1px solid ${({ theme }) => theme.colors.white[500]};
    ${({ theme, position }) => {
      if (position) {
        if (Array.isArray(position)) {
          return `
            top: -${theme.spaces[position[0]]};
            right: -${theme.spaces[position[1]]};
          `;
        }
        return `
          top: -${theme.spaces[position]};
          right: -${theme.spaces[position]};
        `;
      }
      return `
        top: -10px;
        right: -10px;
      `;
    }}
    ${({ theme }) => `
      min-width: ${theme.spaces[4]};
      width: ${theme.spaces[4]};
      height: ${theme.spaces[4]};
    `}
    padding: 0;

    font-family: ${({ theme }) => theme.typography.body3.family};
    font-size: ${({ theme }) => theme.typography.body3.size};
    line-height: ${({ theme }) => theme.typography.body3.lineHeight};
    letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  }
`;

export const BadgeStyle = styled.div<HubBadgeProps>`
  ${({ float, content }) => {
    if (float) {
      if (content) {
        return floatStyle;
      }
    } else {
      return allStyle;
    }
    return '';
  }}
`;
