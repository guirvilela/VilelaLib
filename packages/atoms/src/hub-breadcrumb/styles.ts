import styled from 'styled-components';

import { HubBreadcrumbProps, HubBreadcrumbsProps } from './types';

export const BreadcrumbsStyle = styled.div<HubBreadcrumbsProps>`
  display: flex;
  align-items: center;
  overflow: auto;
`;

export const BreadcrumbsPathStyle = styled.div`
  display: flex;
  align-items: center;
`;

export const BreadcrumbsIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.neutral[400]};
  margin: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const BreadcrumbStyle = styled.span<HubBreadcrumbProps>`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme, active }) =>
    active ? theme.colors.primary[500] : theme.colors.neutral[400]};
  transition: color 0.5s;
  font-weight: 600;
  white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : 'pre-wrap')};

  ${({ width }) =>
    width
      ? `
    display: block;
    max-width: ${width};
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  `
      : ''}

  &:hover {
    color: ${({ theme }) => theme.colors.primary[500]};
  }
`;
