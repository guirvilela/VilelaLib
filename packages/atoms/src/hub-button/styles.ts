import { shadeColor } from '@hubert/shared';
import { animations } from '@hubert/styles';
import styled, { css } from 'styled-components';

import { HubButtonProps } from './types';

const ButtonSizesStyle = css<HubButtonProps>`
  width: ${({ theme, full, size, variant }) => {
    if (full) return '100%';
    if (variant === 'icon') {
      switch (size) {
        case 'xs':
          return theme.spaces[5];
        case 'sm':
          return theme.spaces[6];
        case 'md':
          return theme.spaces[7];
        case 'lg':
          return theme.spaces[8];
      }
    }
    return 'auto';
  }};
  height: ${({ theme, size, variant }) => {
    switch (size) {
      case 'xs':
        return theme.spaces[5];
      case 'sm':
        return theme.spaces[6];
      case 'md':
        return theme.spaces[7];
      case 'lg':
        return theme.spaces[8];
      default:
        if (size === '100%') {
          return size;
        }
        if (variant === 'icon' || variant === 'action') {
          return 'auto';
        }
        return theme.spaces[7];
    }
  }};
  padding: ${({ theme, variant, size, noPadding }) => {
    if (variant === 'icon' || variant === 'action' || noPadding) {
      return undefined;
    }
    switch (size) {
      case 'xs':
        return `${theme.spaces[1]} ${theme.spaces[2]}`;
      default:
        return `${theme.spaces[1]} ${theme.spaces[3]}`;
    }
  }};
`;

const ButtonStylesStyle = css<HubButtonProps>`
  ${({
    theme,
    variant,
    color,
    colorLevel,
    bgColor,
    bgColorLevel,
    borderColor,
    borderColorLevel
  }) => {
    const hash =
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : theme.colors.primary[500];
    const bgHash =
      bgColor && bgColorLevel
        ? theme.colors[bgColor][bgColorLevel]
        : bgColor && theme.colors[bgColor]
        ? theme.colors[bgColor][500]
        : bgColor
        ? bgColor
        : 'transparent';
    const borderHash =
      borderColor && borderColorLevel
        ? theme.colors[borderColor][borderColorLevel]
        : borderColor && theme.colors[borderColor]
        ? theme.colors[borderColor][500]
        : borderColor
        ? borderColor
        : hash;

    switch (variant) {
      case 'outline':
        return `
          background-color: ${bgHash};
          color: ${hash};
          border: 1.5px solid ${borderHash};
        `;
      case 'ghost':
      case 'icon':
      case 'action':
        return `
          background-color: ${bgHash};
          color: ${hash};
        `;
      default:
        return `
          background-color: ${hash};
          color: ${theme.colors.white[500]};
        `;
    }
  }}
`;

const ButtonHoverStyle = css<HubButtonProps>`
  &:hover:not(:disabled) {
    ${({ theme, variant, color, bgHover }) => {
      const hash =
        color === 'white'
          ? theme.colors.neutralGrey[300]
          : color && theme.colors[color]
          ? shadeColor(theme.colors[color][500], -5)
          : color
          ? color
          : theme.colors.neutral[600];

      if (bgHover) {
        switch (variant) {
          case 'default':
            return `
              background-color: ${hash};
            `;
          default:
            return `
              background-color: ${theme.colors.neutralGrey[100]};
            `;
        }
      }

      switch (variant) {
        case 'outline':
          return `
            background-color: ${theme.colors.neutralGrey[100]};
          `;
        case 'ghost':
        case 'icon':
        case 'action':
          return `
            color: ${hash};
          `;
        default:
          return `
            background-color: ${hash};
          `;
      }
    }}
  }
`;

const ButtonActiveStyle = css<HubButtonProps>`
  &:not(:disabled) {
    ${({ theme, color, colorLevel }) => {
      const hash =
        color && colorLevel
          ? theme.colors[color][colorLevel]
          : color && theme.colors[color]
          ? theme.colors[color][500]
          : color
          ? color
          : theme.colors.primary[500];
      return `
        outline: 1px solid ${hash};
      `;
    }}
  }
`;

const ButtonDisabledStyle = css<HubButtonProps>`
  &:disabled {
    ${({ theme, variant }) => {
      switch (variant) {
        case 'outline':
          return `
            border-color: ${theme.colors.primary[100]};
            color: ${theme.colors.primary[100]};
          `;
        case 'ghost':
        case 'icon':
        case 'action':
          return `
            color: ${theme.colors.primary[100]};
          `;
        default:
          return `
            background-color: ${theme.colors.neutral[300]};
          `;
      }
    }}
  }
`;

const ButtonCommonStyle = css<HubButtonProps>`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: ${({ theme }) => theme.spaces[2]};
  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || 1]};
  border: none;
  font-weight: 600;
  transition: background-color, outline 0.5s;
  cursor: ${({ cursor, dragTrigger }) =>
    dragTrigger ? 'grab' : cursor || 'pointer'};
  white-space: nowrap;
  text-decoration: none;
`;

export const ButtonLoadingStyle = styled.div`
  width: ${({ theme }) => theme.spaces[5]};
  height: ${({ theme }) => theme.spaces[5]};
  border-radius: 50%;
  animation: 1s linear infinite ${() => animations.spin},
    ${() => animations.fadein} 1s ease;
`;

const ButtonShared = css<HubButtonProps>`
  ${ButtonCommonStyle}
  ${ButtonSizesStyle}
  ${ButtonStylesStyle}
  ${ButtonDisabledStyle}
  ${({ noHover }) => (noHover ? '' : ButtonHoverStyle)}
  ${({ active }) => (active ? ButtonActiveStyle : '')}

  ${({ clickable, disabled }) => {
    if (clickable === false) {
      return `
        pointer-events: none !important;
        cursor: not-allowed !important;
      `;
    }
    if (disabled) {
      return `
        cursor: not-allowed !important;
      `;
    }
    return '';
  }}

  ${ButtonLoadingStyle} {
    ${({ theme, variant, color, colorLevel }) => {
      switch (variant) {
        case 'outline':
        case 'ghost':
        case 'icon':
        case 'action':
          return `
              border: 3px solid ${
                color && theme.colors[color]
                  ? theme.colors[color][100]
                  : color
                  ? color
                  : theme.colors.primary[100]
              };
              border-top: 3px solid ${
                color && colorLevel
                  ? theme.colors[color][colorLevel]
                  : color && theme.colors[color]
                  ? theme.colors[color][500]
                  : color
                  ? color
                  : theme.colors.primary[300]
              };
            `;
        default:
          return `
              border: 3px solid rgba(255, 255, 255, 0.5);
              border-top: 3px solid ${
                color && colorLevel
                  ? theme.colors[color][colorLevel]
                  : color && theme.colors[color]
                  ? theme.colors[color][500]
                  : color
                  ? color
                  : theme.colors.white[500]
              };
            `;
      }
    }}
  }
`;

export const ButtonStyle = styled.button<HubButtonProps>`
  ${ButtonShared}
`;

export const ButtonLinkStyle = styled.a<HubButtonProps>`
  ${ButtonShared}
`;
