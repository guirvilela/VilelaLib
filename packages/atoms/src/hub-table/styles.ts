import { Align, Sort, animations } from '@hubert/styles';
import styled, { css } from 'styled-components';

import { HubTableColumn, HubTableRowConfig } from './types';

export const TableStyle = styled.table`
  border-collapse: collapse;
  table-layout: fixed;
  width: 100%;
  ${({ width }) => (width ? `min-width: ${width};` : '')}
`;

export const TableHeadColStyle = styled.th<
  Pick<
    HubTableColumn<unknown>,
    'width' | 'minWidth' | 'sticky' | 'printable'
  > & { index }
>`
  width: ${({ width }) => width || '100%'};
  ${({ minWidth }) => (minWidth ? `min-width: ${minWidth};` : '')};
  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};

  position: sticky;
  top: 0;
  z-index: 2;

  ${({ index, sticky }) =>
    sticky &&
    `
      &:nth-child(${index + 1}) {
        position: sticky;
        left: 0;
        z-index: 3;
      }
  `}
  ${({ printable }) =>
    printable === false &&
    `
    @media print {
      display: none;
    }
  `}
`;

export const TableBodyColStyle = styled.td<
  Pick<
    HubTableColumn<unknown>,
    'width' | 'minWidth' | 'sticky' | 'printable'
  > & {
    index: number;
    foot?: boolean;
  }
>`
  width: ${({ width }) => width || '100%'};
  ${({ minWidth }) => (minWidth ? `min-width: ${minWidth};` : '')};
  ${({ theme, index, sticky }) =>
    sticky &&
    `
    &:nth-child(${index + 1}) {
      position: sticky;
      left: 0;
      z-index: 1;
      background-color: ${theme.colors.white[500]};
    }
  `}
  ${({ printable }) =>
    printable === false &&
    `
    @media print {
      display: none;
    }
  `}
  ${({ theme, foot }) =>
    foot &&
    `
    position: sticky;
    bottom: 0;
    z-index: 2;
    background-color: ${theme.colors.neutralGrey['A100']};
    font-weight: 600;
  `}
`;

export const TableHeadRowStyle = styled.tr<HubTableRowConfig>`
  ${({ theme, height }) => `
    ${TableHeadColStyle} {
      background-color: ${theme.colors.neutral[200]};
      color: ${theme.colors.neutral[800]};
      font-family: ${theme.typography.body2.family};
      font-size: ${theme.typography.body2.size};
      line-height: ${theme.typography.body2.lineHeight};
      letter-spacing: ${theme.typography.body2.letterSpacing};
      font-weight: 600;
      height: ${height || theme.spaces[7]};
    }
  `}
`;

export const TableBodyColInnerStyle = styled.div`
  display: flex;
  align-items: center;
  min-width: 0;
  width: 100%;
  height: 100%;
`;

export const TableBodyColCellStyle = styled.div<
  Pick<HubTableColumn<unknown>, 'nowrap' | 'align' | 'noOverflow'>
>`
  width: 100%;
  text-align: ${({ align }) => align || 'center'};

  ${({ noOverflow, nowrap }) => {
    if (!noOverflow) {
      return `
        overflow: hidden;
        text-overflow: ellipsis;

        ${
          nowrap
            ? `white-space: nowrap;`
            : `
        display: -webkit-box;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
      `
        }
      `;
    }
    return '';
  }}

  ${({ theme }) => `
    font-family: ${theme.typography.body2.family};
    font-size: ${theme.typography.body2.size};
    line-height: ${theme.typography.body2.lineHeight};
    letter-spacing: ${theme.typography.body2.letterSpacing};
  `}
`;

const RowBlink = css`
  animation: 3s linear ${() => animations.blink};
`;

export const TableBodyRowStyle = styled.tr<
  HubTableRowConfig & { borderless?: boolean; clickable?: boolean }
>`
  ${({ blink }) => (blink ? RowBlink : '')}

  ${({ theme, borderless, border, borderColor, accentColor }) =>
    borderless || border || borderColor || accentColor
      ? ''
      : `border-bottom: 1px solid ${theme.colors.neutral[300]};`}

  ${TableBodyColInnerStyle} {
    padding: ${({ theme }) => `0 ${theme.spaces[2]}`};

    ${({ theme, height, color, colorLevel }) => `
      background-color: ${
        color && colorLevel
          ? theme.colors[color][colorLevel]
          : color
          ? theme.colors[color][500]
          : undefined
      };
      height: ${height || theme.spaces[7]};
    `}
  }

  ${TableBodyColStyle} {
    ${({
      theme,
      border,
      borderStyle,
      borderColor,
      borderColorLevel,
      accentColor,
      accentColorLevel,
      borderRadius
    }) => {
      const borderHash =
        borderColor && borderColorLevel
          ? theme.colors[borderColor][borderColorLevel]
          : borderColor
          ? theme.colors[borderColor][500]
          : theme.colors.neutral[300];

      const accentHash =
        accentColor && accentColorLevel
          ? theme.colors[accentColor][accentColorLevel]
          : accentColor
          ? theme.colors[accentColor][500]
          : undefined;

      const borderConfig = `${border || '1px'} ${
        borderStyle || 'solid'
      } ${borderHash}`;

      const radius = theme.spaces[borderRadius || '1'];

      return border || borderColor || accentColor
        ? `
        ${TableBodyColInnerStyle} {
          border-top: ${borderConfig};
          border-bottom: ${borderConfig};
        }

        &:first-of-type ${TableBodyColInnerStyle} {
          border: ${borderConfig};
          border-right: none;
          border-top-left-radius: ${radius};
          border-bottom-left-radius: ${radius};
          ${accentHash ? `border-left: 8px solid ${accentHash};` : ''}
        }

        &:last-of-type ${TableBodyColInnerStyle} {
          border: ${borderConfig};
          border-left: none;
          border-top-right-radius: ${radius};
          border-bottom-right-radius: ${radius};
        }
      `
        : '';
    }}

    padding-top: ${({ theme, gap }) => (gap ? theme.spaces[gap] : 0)};

    ${({ clickable }) =>
      clickable
        ? `
      cursor: pointer;
    `
        : ''}
  }
`;

export const TableHeadOptionsStyle = styled.div<{ align: Align }>`
  display: flex;
  align-items: center;
  text-align: ${({ align }) => align || 'center'};
  justify-content: ${({ align }) => align || 'center'};
  gap: ${({ theme }) => theme.spaces[2]};
`;

export const TableHeadSortStyle = styled.button<{ state: Sort }>`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: transparent;
  border: none;
  transform: ${({ state }) => `rotate(${state === 'asc' ? '180deg' : '0deg'})`};
  cursor: pointer;

  path:first-of-type {
    fill: ${({ theme }) => theme.colors.neutralGrey[600]} !important;
  }
`;

export const TableContainerStyle = styled.div<{
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref?: any;
  minHeight?: string;
  height?: string;
}>`
  width: 100%;
  overflow: auto;
  max-height: ${({ height }) => height || '100%'};
  ${({ minHeight }) => (minHeight ? `min-height: ${minHeight};` : '')}

  @media print {
    max-height: unset;
    height: auto;
    overflow: unset;

    ${TableHeadRowStyle},
    ${TableBodyColInnerStyle} {
      height: auto;
    }
  }

  ${TableHeadColStyle},
  ${TableBodyColStyle} {
    @media print {
      width: unset;
    }
  }

  ${TableBodyColCellStyle} {
    @media print {
      display: block;
      overflow: unset;
      text-overflow: unset;
    }
  }

  ${TableHeadSortStyle} {
    @media print {
      display: none;
    }
  }
`;

export const TableLoadMoreStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  z-index: 3;

  bottom: 0;
  left: 0;
  right: 0;
  background-color: ${({ theme }) => theme.colors.white[500]};
  color: ${({ theme }) => theme.colors.neutralGrey[600]};
  padding: ${({ theme }) => theme.spaces[2]} 0;
  box-shadow: 0px -10px 24px -10px rgba(0, 0, 0, 0.2);
  font-weight: 600;
  border-bottom-left-radius: ${({ theme }) => theme.spaces[3]};
  border-bottom-right-radius: ${({ theme }) => theme.spaces[3]};
  animation: 0.5s ease ${() => animations.fadein};
`;
