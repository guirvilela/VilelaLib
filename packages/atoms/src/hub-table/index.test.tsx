import { render } from '@testing-library/react';
import React from 'react';

import { wrapse } from '@hubert/shared';
import { ThemeProvider } from '@hubert/styles';

import { HubTable } from '.';

describe('HubTable component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HubTable columns={[]} data={[]} />, {
      wrapper: wrapse(ThemeProvider)
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
