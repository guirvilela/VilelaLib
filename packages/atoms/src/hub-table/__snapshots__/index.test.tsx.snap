// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubTable component matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  border-collapse: collapse;
  table-layout: fixed;
  width: 100%;
}

.c0 {
  width: 100%;
  overflow: auto;
  max-height: 100%;
}

@media print {
  .c0 {
    max-height: unset;
    height: auto;
    overflow: unset;
  }
}

@media print {

}

@media print {

}

@media print {

}

<div
    class="c0"
  >
    <table
      class="c1"
    >
      <thead />
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
